### PR TITLE
Moved symbol finding to OrbitService::Utils

### PR DESCRIPTION
**Pull request by antonrohr**
This cuts the last dependency of OrbitService to OrbitCore.

### CODE REVIEW

File: OrbitCore/SymbolHelper.h [hunk 1]
```diff
@@ -14,25 +14,17 @@
 class SymbolHelper {
  public:
   SymbolHelper();
-  SymbolHelper(std::vector<std::string> collector_symbol_directories,
-               std::vector<std::string> symbols_file_directories)
-      : collector_symbol_directories_(std::move(collector_symbol_directories)),
-        symbols_file_directories_(std::move(symbols_file_directories)){};
+  SymbolHelper(std::vector<std::string> symbols_file_directories)
```
Comment (dimitry-) [inline OrbitCore/SymbolHelper.h:17]: explicit?

File: OrbitService/Utils.cpp [hunk 2]
```diff
@@ -177,6 +178,82 @@ ErrorMessageOr<std::string> ReadFileToString(const std::filesystem::path& file_n
       std::string{std::istreambuf_iterator<char>{file_stream}, std::istreambuf_iterator<char>{}});
 }
 
+ErrorMessageOr<Path> FindSymbolsFilePath(const Path& module_path,
+                                         const std::vector<Path>& search_directories) {
+  OUTCOME_TRY(module_elf_file, ElfFile::Create(module_path.string()));
+  if (module_elf_file->HasSymtab()) {
+    return outcome::success(module_path);
```
Comment (dimitry-) [inline OrbitService/Utils.cpp:185]: just module_path

File: OrbitService/Utils.cpp [hunk 3]
```diff
@@ -177,6 +178,82 @@ ErrorMessageOr<std::string> ReadFileToString(const std::filesystem::path& file_n
...
+    }
+
+    return outcome::success(symbols_path);
+  }
+
+  std::string error_message_for_client{absl::StrFormat(
+      "Unable to find debug symbols on the instance for module \"%s\". ", module_path)};
+  if (!error_messages.empty()) {
+    absl::StrAppend(&error_message_for_client, "\nDetails:\n", absl::StrJoin(error_messages, "\n"));
+  }
```
Comment (dimitry-) [inline OrbitService/Utils.cpp:253]: nice!

File: OrbitGrpcProtos/services.proto [hunk 4]
```diff
@@ -50,7 +50,6 @@ message GetProcessMemoryResponse {
 
 message GetDebugInfoFileRequest {
```
Comment (dimitry-) [inline OrbitGrpcProtos/services.proto:51]: reserved 2;

File: OrbitService/Utils.cpp [hunk 5]
```diff
@@ -177,6 +178,82 @@ ErrorMessageOr<std::string> ReadFileToString(const std::filesystem::path& file_n
...
+      std::string error_message = absl::StrFormat(
+          "Potential symbols file \"%s\" has a different build id than the module requested by the "
+          "client. \"%s\" != \"%s\"",
+          symbols_path, build_id, module_elf_file->GetBuildId());
+      LOG("%s", error_message);
+      error_messages.emplace_back("* " + std::move(error_message));
+      continue;
+    }
+
+    return outcome::success(symbols_path);
```
Comment (dimitry-) [inline OrbitService/Utils.cpp:246]: also drop outcome::success

**Review by dimitry- — APPROVED** [review]
otherwise lgtm

File: OrbitService/Utils.cpp (hunk 5 above)
Comment (antonrohr) [reply OrbitService/Utils.cpp]: done

File: OrbitGrpcProtos/services.proto (hunk 4 above)
Comment (antonrohr) [reply OrbitGrpcProtos/services.proto:51]: done

File: OrbitService/Utils.cpp (hunk 3 above)
Comment (antonrohr) [reply OrbitService/Utils.cpp:252]: thanks :)

File: OrbitService/Utils.cpp (hunk 2 above)
Comment (antonrohr) [reply OrbitService/Utils.cpp]: done

File: OrbitCore/SymbolHelper.h (hunk 1 above)
Comment (antonrohr) [reply OrbitCore/SymbolHelper.h]: done

File: OrbitGrpcProtos/services.proto [hunk 6]
```diff
@@ -50,6 +50,7 @@ message GetProcessMemoryResponse {
...
   string module_path = 1;
+  reserved 2;
```
Comment (dimitry-) [inline OrbitGrpcProtos/services.proto:53]: lets have reserved list in the first line, this way we won't need to move it around

Comment (dimitry-) [reply OrbitGrpcProtos/services.proto:53]: just to clarify - this is a list so if we have 2 reserved values it becomes reserved 1, 2;

Comment (antonrohr) [reply OrbitGrpcProtos/services.proto:53]: okay, makes sense - done, will merge now

**Pull request merged**